Add C shim for at::pad and fix some typos

janeyx99 opened the pull request:
As stated, we would like a pad shim to support custom ops wanting to build in an ABI stable manner.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #155226



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

janeyx99 committed:
Add C shim for at::pad and fix some typos

[ghstack-poisoned]

janeyx99 committed:
Update on "Add C shim for at::pad and fix some typos"

As stated, we would like a pad shim to support custom ops wanting to build in an ABI stable manner.




[ghstack-poisoned]

janeyx99 committed:
Update on "Add C shim for at::pad and fix some typos"

As stated, we would like a pad shim to support custom ops wanting to build in an ABI stable manner.




[ghstack-poisoned]

desertfire reviewed (approved):
Let's add a test in test_aot_inductor.py

janeyx99 committed:
Update on "Add C shim for at::pad and fix some typos"

As stated, we would like a pad shim to support custom ops wanting to build in an ABI stable manner.




[ghstack-poisoned]

janeyx99 committed:
Update on "Add C shim for at::pad and fix some typos"

As stated, we would like a pad shim to support custom ops wanting to build in an ABI stable manner.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]
